module-2/backend/week_four.md

```diff
@@ -10,14 +10,49 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 * What is a cookie?
+
+stored data from a web site like what's in your shopping cart or links you have clicked on
+
 * What’s the difference between a session and a cookie?
+
+a session in your log in and specific to your user. a cookie is saved data outside of the session
+
 * What’s a flash and when do you want to use flashes?
+
+a flash is how you can display a temporary message like: 'successful login', 'logged out', etc.
+
 * Why do people say “HTTP is stateless”?
+
+because the cookies and sessions is what hold state on your computer
+
 * What’s authentication? Explain.
+
+authentication is how you can tell a user is who they say they are. i.e. email, username, password
+
 * What’s the difference between authentication and authorization?
+
+authorization is the power to do something you set. for example, if you dont want users to be able to delete gifs then you would create an admin role and only give the admin the ability to do it.
```

Comment:
authentication = i am who i say i am vs authorization = i have permission to be here